// hume/dist/serialization/resources/empathicVoice/types/WebhookEventChatEnded.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookEventChatEnded = void 0;
const core = __importStar(require("../../../../core"));
const WebhookEventChatStatus_1 = require("./WebhookEventChatStatus");
const WebhookEventBase_1 = require("./WebhookEventBase");
exports.WebhookEventChatEnded = core.serialization
    .object({
    callerNumber: core.serialization.property("caller_number", core.serialization.string().optional()),
    customSessionId: core.serialization.property("custom_session_id", core.serialization.string().optional()),
    durationSeconds: core.serialization.property("duration_seconds", core.serialization.number()),
    endReason: core.serialization.property("end_reason", WebhookEventChatStatus_1.WebhookEventChatStatus),
    endTime: core.serialization.property("end_time", core.serialization.number()),
    eventName: core.serialization.property("event_name", core.serialization.stringLiteral("chat_ended").optional()),
})
    .extend(WebhookEventBase_1.WebhookEventBase);
